src/emu/http.cpp

```diff
@@ -261,57 +263,38 @@ struct websocket_connection_impl : public http_manager::websocket_connection {
 	}
 };
 
-http_manager::http_manager(bool active, short port, const char *root)
+http_manager::http_manager(bool active, const char *address, short port, const char *root)
```

Comment:
Would it be better to look for things that are path separators but are not slashes?  That in itself is probably a warning sign.

Reply:
I'm not following? 

when trying to verify a filename, we need to check for anything that navigates upwards among the filesystem path to try to escape the web server root. Filenames of course have to be given with the path separators that the platform on which the program is running uses; on Unix-esque systems, the path separator _is_ a slash ('/') , so we can't exclude '/' form being searched for as a path separator.